api/v1alpha1/releaseserviceconfig_types.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const ReleaseServiceConfigResourceName string = "config"
+
+// ReleaseServiceConfigSpec defines the desired state of ReleaseServiceConfig.
+type ReleaseServiceConfigSpec struct {
+	// Debug is the boolean that specifies whether or not the Release Service should run
+	// in debug mode
+	// +optional
+	Debug bool `json:"debug,omitempty"`
+}
+
+// ReleaseServiceConfigStatus defines the observed state of ReleaseServiceConfig.
+type ReleaseServiceConfigStatus struct {
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// ReleaseServiceConfig is the Schema for the releaseserviceconfigs API
+type ReleaseServiceConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ReleaseServiceConfigSpec   `json:"spec,omitempty"`
+	Status ReleaseServiceConfigStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// ReleaseServiceConfigList contains a list of ReleaseServiceConfig
+type ReleaseServiceConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ReleaseServiceConfig `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&ReleaseServiceConfig{}, &ReleaseServiceConfigList{})
+}
+
+// AreDebugOptionsValid checks whether the bundle value is empty if Debug mode is False.
+func (r *ReleaseServiceConfig) AreDebugOptionsValid(bundle string) bool {
+	return r.Spec.Debug || bundle != ""
```

Comment:
I don't think this belongs here. This is adapter/tekton logic.

Reply:
Why? Why does the adapter care about how the ReleaseServiceConfig makes decisions? Also your previous comment was 
`this function should be renamed to something better. Maybe something like localPipelinesAllowed for readability or AreLocalPipelinesAllowed to start with Is/Are like in the Release type.`
which made it sound like this function is fine here

Reply:
Because before it was just an accesor. Now it's making decisions and a config file/resource should not make decisions. We have the adapter for that.

Reply:
I'm not following. All that changed from the previous version was a field in the spec was renamed. The logic is the same as when this function was okay

Reply:
Maybe I didn't reached that far. In any case, I don't think that logic should be in here.